src/scene/mesh.js

```diff
@@ -555,23 +555,51 @@ Object.assign(pc, function () {
         }
     });
 
-    var InstancingData = function (numObjects, dynamic, instanceSize) {
-        instanceSize = instanceSize || 16;
-        this.buffer = new Float32Array(numObjects * instanceSize);
+    // internal data structure used to store data used by hardware instancing
+    var InstancingData = function (numObjects) {
         this.count = numObjects;
+        this.vertexBuffer = null;
         this.offset = 0;
-        this.usage = dynamic ? pc.BUFFER_DYNAMIC : pc.BUFFER_STATIC;
-        this._buffer = null;
     };
 
-    Object.assign(InstancingData.prototype, {
-        update: function () {
-            if (this._buffer) {
-                this._buffer.setData(this.buffer);
+     /**
+      * @function
+      * @name pc.MeshInstance#setInstancing
+      * @description Sets up {@link pc.MeshInstance} to be rendered using Hardware Instancing.
+      * @param {pc.VertexBuffer|null} vertexBuffer - Vertex buffer to hold per-instance vertex data (usually world matrices).
+      * Pass null to turn off hardware instancing.
+      */
+    Object.assign(MeshInstance.prototype, {
+        setInstancing: function (vertexBuffer) {

```

Comment:
Can you define this in the previous `Object.assign(MeshInstance.prototype, {` block?